hal/src/rtl872x/gpio_hal.cpp

```diff
@@ -141,18 +141,19 @@ int hal_gpio_configure(hal_pin_t pin, const hal_gpio_config_t* conf, void* reser
         GPIO_InitStruct.GPIO_Pin = rtlPin;
 
         switch (mode) {
-            case OUTPUT:
-            case OUTPUT_OPEN_DRAIN: {
+            case OUTPUT: {
                 GPIO_InitStruct.GPIO_Mode = GPIO_Mode_OUT;
                 GPIO_InitStruct.GPIO_PuPd = GPIO_PuPd_NOPULL;
                 break;
             }
-            case INPUT: {
+            case INPUT:
+            case OUTPUT_OPEN_DRAIN: {
                 GPIO_InitStruct.GPIO_Mode = GPIO_Mode_IN;
                 GPIO_InitStruct.GPIO_PuPd = GPIO_PuPd_NOPULL;
                 break;
             }
-            case INPUT_PULLUP: {
+            case INPUT_PULLUP:
+            case OUTPUT_OPEN_DRAIN_PULLUP: {

```

Comment:
If `conf->set_value == true` and `conf->value == 0` we need to configure as `GPIO_Mode_OUT` with `DR` register bit already set to 0 for the pin.